go/vt/vttablet/tabletserver/state_manager_test.go

```diff
@@ -457,9 +456,16 @@ func TestStateManagerCheckMySQL(t *testing.T) {
 	err := sm.SetServingType(topodatapb.TabletType_PRIMARY, testNow, StateServing, "")
 	require.NoError(t, err)
 
+	sm.te = &delayedTxEngine{}
 	sm.qe.(*testQueryEngine).failMySQL = true
 	order.Set(0)
 	sm.CheckMySQL()
+	// We know checkMySQL will take atleast 50 milliseconds since txEngine.Close has a sleep in the test code
+	time.Sleep(10 * time.Millisecond)
+	// this asserts that checkMySQL is running
+	assert.EqualValues(t, 0, sm.checkMySQLThrottler.Size())
```

Comment:
this has been slightly changed here, because the function `isCheckMySQLRunning` isn't available in this or previous releases